# Assignment 1
### Understanding Uncertainty
### Due 9/5

Shiraz Robinson II

1. Create a new public repo on Github under your account. Include a readme file.
2. Clone it to your machine. Put this file into that repo.
3. Use the following function to download the example data for the course:

In [1]:
def download_data(force=False):
    """Download and extract course data from Zenodo."""
    import urllib.request, zipfile, os
    
    zip_path = 'data.zip'
    data_dir = 'data'
    
    if not os.path.exists(zip_path) or force:
        print("Downloading course data")
        urllib.request.urlretrieve(
            'https://zenodo.org/records/16954427/files/data.zip?download=1',
            zip_path
        )
        print("Download complete")
    else:
        print("Download file already exists")
        
    if not os.path.exists(data_dir) or force:
        print("Extracting data files...")
        with zipfile.ZipFile(zip_path, 'r') as zip_ref:
            zip_ref.extractall(data_dir)
        print("Data extracted")
    else:
        print("Data directory already exists")

download_data()

Download complete
Extracting data files...
Data extracted


4. Open one of the datasets using Pandas:
    1. `ames_prices.csv`: Housing characteristics and prices
    2. `college_completion.csv`: Public, nonprofit, and for-profit educational institutions, graduation rates, and financial aid
    3. `ForeignGifts_edu.csv`: Monetary and in-kind transfers from foreign entities to U.S. educational institutions
    4. `iowa.csv`: Liquor sales in Iowa, at the transaction level
    5. `metabric.csv`: Cancer patient and outcome data
    6. `mn_police_use_of_force.csv`: Records of physical altercations between Minnessota police and private citizens
    7. `nhanes_data_17_18.csv`: National Health and Nutrition Examination Survey
    8. `tuna.csv`: Yellowfin Tuna Genome (I don't recommend this one; it's just a sequence of G, C, A, T )
    9. `va_procurement.csv`: Public spending by the state of Virginia

In [3]:
import pandas as pd

df = pd.read_csv('./data/nhanes_data_17_18.csv')

df.head()

/var/folders/n2/lt2mc7z901q9k1mx_2s6ypb00000gn/T/ipykernel_82619/2342757229.py:3: DtypeWarning: Columns (142) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('./data/nhanes_data_17_18.csv')


SEQN GeneralHealthCondition  EverBreastfedOrFedBreastmilk  \
0  93703.0                    NaN                           1.0   
1  93704.0                    NaN                           1.0   
2  93705.0                   Good                           NaN   
3  93706.0              Very good                           NaN   
4  93707.0                   Good                           NaN   

   AgeStoppedBreastfeedingdays  AgeFirstFedFormuladays  \
0                        273.0                     1.0   
1                         60.0                     3.0   
2                          NaN                     NaN   
3                          NaN                     NaN   
4                          NaN                     NaN   

   AgeStoppedReceivingFormuladays  AgeStartedOtherFoodbeverage  \
0                           365.0                        152.0   
1                           365.0                        126.0   
2                             NaN                          NaN   
3                             NaN                          NaN   
4                             NaN                          NaN   

   AgeFirstFedMilkdays TypeOfMilkFirstFedWholeMilk  TypeOfMilkFirstFed2Milk  \
0                365.0       Whole or regular milk                      NaN   
1                365.0       Whole or regular milk                      NaN   
2                  NaN                         NaN                      NaN   
3                  NaN                         NaN                      NaN   
4                  NaN                         NaN                      NaN   

   ...  DaysSmokedCigsDuringPast30Days  AvgCigarettesdayDuringPast30Days  \
0  ...                             NaN                               NaN   
1  ...                             NaN                               NaN   
2  ...                             NaN                               NaN   
3  ...                             NaN                               NaN   
4  ...                             NaN                               NaN   

   TriedToQuitSmoking  TimesStoppedSmokingCigarettes  \
0                 NaN                            NaN   
1                 NaN                            NaN   
2                 NaN                            NaN   
3                 NaN                            NaN   
4                 NaN                            NaN   

  HowLongWereYouAbleToStopSmoking UnitOfMeasureDayweekmonthyear_2_SMQ  \
0                             NaN                                 NaN   
1                             NaN                                 NaN   
2                             NaN                                 NaN   
3                             NaN                                 NaN   
4                             NaN                                 NaN   

  CurrentSelfreportedHeightInches  CurrentSelfreportedWeightPounds  \
0                             NaN                              NaN   
1                             NaN                              NaN   
2                            63.0                            165.0   
3                            68.0                            145.0   
4                             NaN                              NaN   

   TriedToLoseWeightInPastYear  TimesLost10LbsOrMoreToLoseWeight  
0                          NaN                               NaN  
1                          NaN                               NaN  
2                          0.0                  11 times or more  
3                          0.0                             Never  
4                          NaN                               NaN  

[5 rows x 198 columns]

5. Pick two or three variables and briefly analyze them
    - Is it a categorical or numeric variable?
    - How many missing values are there? (`df['var'].isna()` and `np.sum()`)
    - If categorical, tabulate the values (`df['var'].value_counts()`) and if numeric, get a summary (`df['var'].describe()`)

In [9]:
import numpy as np

In [13]:
# Variable #1: GeneralHealthCondition
# Categorical variable
print("Number of missing values:", np.sum(df['GeneralHealthCondition'].isna()))
df['GeneralHealthCondition'].value_counts()

Number of missing values: 2402


GeneralHealthCondition
Good         2454
Very good    1544
Fair or      1175
Excellent     619
Poor?         172
Name: count, dtype: int64

In [14]:
# Variable #2: CurrentSelfreportedWeightPounds
# Numeric variable
print("Number of missing values:", np.sum(df['CurrentSelfreportedWeightPounds'].isna()))
df['CurrentSelfreportedWeightPounds'].describe()

Number of missing values: 2538


count    5828.000000
mean      350.995710
std      1280.908078
min        72.000000
25%       145.000000
50%       171.500000
75%       208.000000
max      9999.000000
Name: CurrentSelfreportedWeightPounds, dtype: float64

In [15]:
# Variable #3: CurrentSelfreportedHeightInches
# Numeric variable
print("Number of missing values:", np.sum(df['CurrentSelfreportedHeightInches'].isna()))
df['CurrentSelfreportedHeightInches'].describe()

Number of missing values: 2590


count    5776.000000
mean      272.224030
std      1414.518896
min        49.000000
25%        63.000000
50%        66.000000
75%        70.000000
max      9999.000000
Name: CurrentSelfreportedHeightInches, dtype: float64

6. What are some questions and prediction tools you could create using these data? Who would the stakeholder be for that prediction tool? What practical or ethical questions would it create? What other data would you want, that are not available in your data?

### Motivation

The NHANES dataset has 198 columns, covering everything from general health condition and diet to smoking, height/weight, and other variables of interest. Grip strength, VO2 max, and Muscular endurance testing via pull-ups are variables of interest, that are not in the dataset. 

### Questions

* How do BMI, age, and physical activty levels predict grip strength (if it were measured)?

* What NHANES variables are strongest predictors of aerobic capacity?

* How do nutrition and body composition interact to affect upper-body endurance?

### Possible Prediction Tools

* Functional Fitness Risk Score

* Chronic Disease Predictor

* Frailty and Aging Index

### Potential Stakeholders

* Public Health Agencies

* Clinicians & Physical Therapists

* Insurance & Policy Makers

* Fitness & Wellness Industry 

* Researchers

### Practical & Ethical Considerations

* Practical:

    * Hard to collect stsndardized VO2 max and pull-up data at scale.

    * Prediction accuracy depends on proxies (NHANES doesn't directly measure these).

    * Ensuring models generalize across diverse populations.

* Ethical:

    * Fairness: Could predictive socres disadvantage groups with lower access to fitness opportunities?

    * Stigma: Risk of labeling individuals "low fitness" in a way that harms confidence or care access.

    * Data Use: How should predictions about frailty or disease risk to be shared, and with whom?

    * Equity in Interventions: If at-risk populations are flagged, are resources guaranteed to help them improve?




7. Commit your work to the repo (`git commit -am 'Finish assignment'` at the command line, or use the Git panel in VS Code). Push your work back to Github and submit the link on Canvas in the assignment tab.